phosphor-auditlog/auditLogMgr.cpp

```diff
@@ -0,0 +1,123 @@
+#include "auditLogMgr.hpp"
+
+#include "auditLogParser.hpp"
+#include "auditLogUtils.hpp"
+
+#include <fcntl.h>
+#include <libaudit.h>
+
+#include <phosphor-logging/lg2.hpp>
+#include <sdeventplus/event.hpp>
+#include <sdeventplus/source/event.hpp>
+#include <xyz/openbmc_project/Common/File/error.hpp>
+
+#include <cstring>
+#include <string>
+
+namespace phosphor
+{
+namespace auditlog
+{
+
+sdbusplus::message::unix_fd ALManager::parseAuditLog()
+{
+#ifdef AUDITLOG_KEEP_JSONFILE
+    ALParseFile parsedFile("/tmp/auditLog.json");
+#else
+    ALParseFile parsedFile;
+#endif
+
+    // Initialize to parse full audit log
+    ALParseAll auditParser(parsedFile);
+
+    lg2::debug("Method ParseAuditLog filePath={FILEPATH}", "FILEPATH",
+               parsedFile.getPath());
+
+    // Parse all the events
+    auditParser.doParse();
+
+    // Get file descriptor to return
+    auto fd = openParseFD(parsedFile);
+
+    if (fd >= 0)
+    {
+        /* Schedule the fd to be closed by sdbusplus when it sends it back over
+         * D-Bus.
+         */
+        sdeventplus::Event event = sdeventplus::Event::get_default();
+        fdCloseParseSource = std::make_unique<sdeventplus::source::Defer>(
+            event, std::bind(std::mem_fn(&ALManager::closeParseFD), this, fd,
```

Comment:
turns out  that the std::mem_fn() wrapper isn't actually needed.  I know it's used a lot in this repo but you don't need to do it here if you don't want to.

Reply:
Can you tell I copied from other code in this repo? :)
I'll remove the use of mem_fn() here.